apps/web/src/layouts/workspace-layout.tsx

```diff
@@ -137,13 +134,21 @@ export const AllWorkspaceContext = ({
 export const CurrentWorkspaceContext = ({
   children,
 }: PropsWithChildren): ReactElement => {
-  const router = useRouter();
   const workspaceId = useAtomValue(rootCurrentWorkspaceIdAtom);
-  useSyncRouterWithCurrentWorkspaceId(router);
-  useSyncRouterWithCurrentPageId(router);
-  useRouterWithWorkspaceIdDefense(router);
   const metadata = useAtomValue(rootWorkspacesMetadataAtom);
   const exist = metadata.find(m => m.id === workspaceId);
+  const router = useRouter();
+  const push = router.push;
+  useEffect(() => {
+    const id = setTimeout(() => {
+      if (!exist) {
+        void push('/');
```

Comment:
This avoids entering a nonexisting workspace. we will fallback to the index page